generator/deploy_openshift_pod.py

```diff
@@ -20,206 +20,361 @@
+        configuration.assert_hostname = False
+        client.Configuration.set_default(configuration)
+        return client.CoreV1Api()
 
-    def get_response_from_pod(self) -> Dict:
+    def get_response_from_pod(self) -> V1Pod:
         """
         Read info from a pod in a namespace
         :return: JSON Dict
         """
         return self.api.read_namespaced_pod(
-            name=self.pod_name, namespace=self.project_name
+            name=self.pod_name, namespace=self.k8s_namespace_name
         )
 
-    def delete_pod(self) -> Dict:
+    def delete_pod(self):
         """
-        Delete pod in a namespace
-        :return: JSON Dict
+        Delete the sandbox pod.
+
+        :return: response from the API server
         """
-        return self.api.delete_namespaced_pod(self.pod_name, self.project_name)
+        try:
+            self.api.delete_namespaced_pod(
+                self.pod_name, self.k8s_namespace_name, body=V1DeleteOptions()
+            )
+        except ApiException as e:
+            logger.debug(e)
+            if e.status != 404:
+                raise
 
-    def create_pod(self) -> Dict:
+    def create_pod(self, pod_manifest: Dict) -> Dict:
         """
         Create pod in a namespace
-        :return: JSON dict
+        :return: response from the API server
         """
         return self.api.create_namespaced_pod(
-            body=self.pod_manifest, namespace=self.project_name
+            body=pod_manifest, namespace=self.k8s_namespace_name
         )
 
-    def is_pod_already_deployed(self):
+    def is_pod_already_deployed(self) -> bool:
         """
-        Check if pod is already deployed
-        :return:
+        Check if the pod is already present.
         """
         try:
-            resp = self.get_response_from_pod()
-            return resp
+            self.get_response_from_pod()
+            return True
         except ApiException as e:
-            logger.info(e.status)
+            logger.debug(e)
             if e.status != 404:
                 logger.error(f"Unknown error: {e!r}")
-                raise GeneratorDeployException
-            else:
-                return None
+                raise GeneratorDeployException(f"Something's wrong with the pod': {e}")
+            return False
+
+    def copy_path_to_pod(self, map: MappedDir):
+        """ copy local path inside the pod """
+        # Copy /tmp/foo local file to /tmp/bar
+        # in a remote pod in namespace <some-namespace>:
+        #   oc cp /tmp/foo <some-namespace>/<some-pod>:/tmp/bar
+        target = f"{self.k8s_namespace_name}/{self.pod_name}:{map.path}"
+        logger.info(f"copy {map.local_dir} -> {target}")
+        # if you're interested: the way openshift does this is that creates a tarball locally
+        # and streams it via exec into the container to a pod process
+        run_command(["oc", "cp", map.local_dir, target])
```

Comment:
I would not call it run_command, I would prefer something like `oc_command`. `run_command` is going to be used by `generator/api.py` for calling from https://github.com/packit-service/packit/blob/master/packit/base_git.py#L228

Reply:
I'm sorry but I have no idea what you mean.

Reply:
I would prefer to merge it and discuss later on.